OWL generator doesn't work for SSSOM

In the SSSOM project, it's not possible to generate OWL. 

```console
$ git clone https://github.com/mapping-commons/sssom
$ cd sssom
$ uv tool install poetry
$ make gen-project
```

Running the above commands (after removing the `--exclude owl` part of the relevant make command) gives the following output:

```python-traceback
make gen-project
poetry run gen-project \
		-d project src/sssom_schema/schema/sssom_schema.yaml && mv project/*.py src/sssom_schema/datamodel
...
WARNING:linkml.generators.owlgen:ignoring equals_string=rdfs literal as unable to tell if literal
WARNING:linkml.generators.owlgen:ignoring equals_string=rdfs literal as unable to tell if literal
WARNING:linkml.generators.owlgen:Null expr in: [None] for http://www.w3.org/2002/07/owl#unionOf None
Traceback (most recent call last):
  File "/Users/cthoyt/dev/sssom/.venv/bin/gen-project", line 10, in <module>
    sys.exit(cli())
             ~~~^^
  File "/Users/cthoyt/dev/sssom/.venv/lib/python3.14/site-packages/click/core.py", line 1161, in __call__
    return self.main(*args, **kwargs)
           ~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "/Users/cthoyt/dev/sssom/.venv/lib/python3.14/site-packages/click/core.py", line 1082, in main
    rv = self.invoke(ctx)
  File "/Users/cthoyt/dev/sssom/.venv/lib/python3.14/site-packages/click/core.py", line 1443, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/cthoyt/dev/sssom/.venv/lib/python3.14/site-packages/click/core.py", line 788, in invoke
    return __callback(*args, **kwargs)
  File "/Users/cthoyt/dev/sssom/.venv/lib/python3.14/site-packages/linkml/generators/projectgen.py", line 249, in cli
    gen.generate(yamlfile, project_config)
    ~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/cthoyt/dev/sssom/.venv/lib/python3.14/site-packages/linkml/generators/projectgen.py", line 140, in generate
    gen_dump = gen.serialize(**serialize_args)
  File "/Users/cthoyt/dev/sssom/.venv/lib/python3.14/site-packages/linkml/generators/owlgen.py", line 231, in serialize
    self.as_graph()
    ~~~~~~~~~~~~~^^
  File "/Users/cthoyt/dev/sssom/.venv/lib/python3.14/site-packages/linkml/generators/owlgen.py", line 204, in as_graph
    self.add_class(cls)
    ~~~~~~~~~~~~~~^^^^^
  File "/Users/cthoyt/dev/sssom/.venv/lib/python3.14/site-packages/linkml/generators/owlgen.py", line 400, in add_class
    pre_node = condition_to_bnode(rule.preconditions)
  File "/Users/cthoyt/dev/sssom/.venv/lib/python3.14/site-packages/linkml/generators/owlgen.py", line 391, in condition_to_bnode
    ixn_listnode = self.transform_class_expression(expr, quantifier_predicate=OWL.someValuesFrom)
  File "/Users/cthoyt/dev/sssom/.venv/lib/python3.14/site-packages/linkml/generators/owlgen.py", line 518, in transform_class_expression
    x = self.transform_class_slot_expression(cls, slot, slot, owltypes)
  File "/Users/cthoyt/dev/sssom/.venv/lib/python3.14/site-packages/linkml/generators/owlgen.py", line 637, in transform_class_slot_expression
    self._complement_of_union_of(
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        [self.transform_class_slot_expression(cls, x, main_slot, owl_types) for x in slot.none_of]
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/Users/cthoyt/dev/sssom/.venv/lib/python3.14/site-packages/linkml/generators/owlgen.py", line 1106, in _complement_of_union_of
    self.graph.add((neg_expr, complement_predicate, self._union_of(exprs)), **kwargs)
    ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/cthoyt/dev/sssom/.venv/lib/python3.14/site-packages/rdflib/graph.py", line 629, in add
    assert isinstance(o, Node), "Object %s must be an rdflib term" % (o,)
           ~~~~~~~~~~^^^^^^^^^
AssertionError: Object None must be an rdflib term
```


this is happening because `self.transform_class_slot_expression(cls, x, main_slot, owl_types)` is allowed to return none, and it's passing a `[None]` into `_complement_of_union_of`. This could have been caught with a type checker.

--- a/packages/linkml/src/linkml/generators/owlgen.py
+++ b/packages/linkml/src/linkml/generators/owlgen.py
@@ -708,24 +708,25 @@ class OwlSchemaGenerator(Generator):
         if slot.all_members:
             owl_exprs.append(self.transform_class_slot_expression(cls, slot.all_members, main_slot, owl_types))
 
-        if slot.any_of:
-            owl_exprs.append(
-                self._union_of(
-                    [self.transform_class_slot_expression(cls, x, main_slot, owl_types) for x in slot.any_of]
-                )
-            )
-        if slot.all_of:
-            owl_exprs.append(
-                self._intersection_of(
-                    [self.transform_class_slot_expression(cls, x, main_slot, owl_types) for x in slot.all_of]
-                )
-            )
-        if slot.none_of:
-            owl_exprs.append(
-                self._complement_of_union_of(
-                    [self.transform_class_slot_expression(cls, x, main_slot, owl_types) for x in slot.none_of]
-                )
-            )
+        def _get_slot_nodes(slot_definition) -> list[BNode | URIRef] | None:
+            if not slot_definition:
+                return None
+            rdflib_nodes = [
+                rdflib_node
+                for slot_expression in slot.any_of
+                if (rdflib_node := self.transform_class_slot_expression(cls, slot_expression, main_slot, owl_types))
+            ]
+            if rdflib_nodes:
+                return rdflib_nodes
+            return None
+
+        if any_of_rdflib_nodes := _get_slot_nodes(slot.any_of):
+            owl_exprs.append(self._union_of(any_of_rdflib_nodes))
+        if all_of_rdflib_nodes := _get_slot_nodes(slot.all_of):
+            owl_exprs.append(self._intersection_of(all_of_rdflib_nodes))
+        if none_of_rdflib_nodes := _get_slot_nodes(slot.none_of):
+            owl_exprs.append(self._complement_of_union_of(none_of_rdflib_nodes))
+
         if slot.exactly_one_of:
             disj_exprs = []
             for i, operand in enumerate(slot.exactly_one_of):

